clients/fides-js/src/lib/events.ts

```diff
@@ -84,6 +84,12 @@ export const dispatchFidesEvent = (
         .consentMethod as FidesEventExtraDetails["consentMethod"],
       ...extraDetails,
     };
+    if (!(extraDetails?.trigger as FidesEventDetailsTrigger)?.origin) {
+      constructedExtraDetails.trigger = {
+        ...(constructedExtraDetails.trigger as FidesEventDetailsTrigger),
+        ...({ origin: "fides" } as FidesEventDetailsTrigger),
+      } as FidesEventDetailsTrigger;
+    }
```

Comment:
This is all just fancy way of setting origin to "fides" if one wasn't provided, while also preserving any other trigger details that might have been provided.